specifyweb/frontend/js_src/lib/components/FormEditor/fetchAllViews.ts

```diff
@@ -78,7 +82,14 @@ const augmentDatabaseView = (
       ? userInformation.availableCollections.find(
           ({ id }) => id === view.collectionId
         )?.collectionName
-      : undefined) ?? camelToHuman(view.viewsetLevel),
+      : undefined) ?? camelToHuman(view.viewsetName),
+  disciplineId: strictParseResourceUrl(
+    defined(
+      userInformation.availableCollections.find(
+        ({ id }) => id === view.collectionId
+      )?.discipline
+    )
+  ).at(1) as number,
```

Comment:
instead of `as number`, could you do `!`?
i.e whenever possible pick the least powerful type assertion so that you are asserting less, and letting TypeScript type-check more

Reply:
or you could replace `.at(1)!` with [1].
`.at()` is useful when:
- you need negative indexes `-1` (last item)
- you want typescript to force you to double check that item is not undefined

but in this case, you know that strictParseResourceUrl always returns a tuple of size 2, so getting element at index `[1]` is safe, so you don't need to use .at()